### ISSUE TITLE
systemd-journald crash with a coredump

### ISSUE THREAD

**Issue by cgeoffroy** [coreos/bugs#334]
I got an unexpected crash with the systemd-journal service on CoreOs version 607.0.0 .
This abrupt crash leads to a restart of the docker service (and all my containers).

The only piece of information related to this crash is:
```
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd-coredump[1208]: Detected coredump of the journal daemon itself, diverted to /var/lib/systemd/coredump/core.systemd-journal.0.764a20295f824802bb8bf929af8987da.393.142969
```
I uploaded the corresponding dump there: [(dropbox) core.systemd-journal.0.764a20295f824802bb8bf929af8987da.393.1429693860000000.xz]( https://www.dropbox.com/s/4wtrxx9ez01z685/core.systemd-journal.0.764a20295f824802bb8bf929af8987da.393.1429693860000000.xz?dl=0). But the 'systemd-journald' executable was compiled without debug symbols so i wasn't able to retrieve a human readable stack trace.

---

Here are the journalctl logs:
```
Apr 22 09:09:24 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:24Z" level="info" msg="+job container_inspect(fraunhoferiais_se_3d-map-tiles_latest)"
Apr 22 09:09:24 dhub-generated-panamax.novalocal dockerd[563]: No such container: fraunhoferiais_se_3d-map-tiles_latest
Apr 22 09:09:24 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:24Z" level="info" msg="-job container_inspect(fraunhoferiais_se_3d-map-tiles_latest) = ERR (1)"
Apr 22 09:09:24 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:24Z" level="error" msg="Handler for GET /containers/{name:.*}/json returned error: No such container: fraunhoferiais_se_3d-map-tiles_latest
Apr 22 09:09:24 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:24Z" level="error" msg="HTTP Error: statusCode=404 No such container: fraunhoferiais_se_3d-map-tiles_latest"
Apr 22 09:09:24 dhub-generated-panamax.novalocal docker[677]: I, [2015-04-22T09:09:24.874998 #40]  INFO -- : Completed 200 OK in 18ms (Views: 16.9ms | ActiveRecord: 0.4ms)
Apr 22 09:09:24 dhub-generated-panamax.novalocal docker[703]: I, [2015-04-22T09:09:24.877602 #10]  INFO -- : Completed 200 OK in 29ms (Views: 0.4ms)
Apr 22 09:09:29 dhub-generated-panamax.novalocal docker[1044]: 2103b00b3fdf: Download complete
Apr 22 09:09:29 dhub-generated-panamax.novalocal docker[1044]: 323a3326d61a: Pulling metadata
Apr 22 09:09:29 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:29Z" level="debug" msg="http://193.52.45.131:8080/v1/images/323a3326d61a304c849b95259ea8aeacf47aef7a4c42ebdb5b1212ce33e141e3/json -- HEADER
Apr 22 09:09:30 dhub-generated-panamax.novalocal docker[1044]: 323a3326d61a: Pulling fs layer
Apr 22 09:09:30 dhub-generated-panamax.novalocal dockerd[563]: time="2015-04-22T09:09:30Z" level="debug" msg="http://193.52.45.131:8080/v1/images/323a3326d61a304c849b95259ea8aeacf47aef7a4c42ebdb5b1212ce33e141e3/layer -- HEADE
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd-journal[1216]: Permanent journal is using 24.0M (max allowed 500.0M, trying to leave 4.0G free of 32.4G available → current limit 500.0M).
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd-coredump[1208]: Detected coredump of the journal daemon itself, diverted to /var/lib/systemd/coredump/core.systemd-journal.0.764a20295f824802bb8bf929af8987da.393.142969
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd-journal[1216]: Journal started
Apr 22 09:09:44 dhub-generated-panamax.novalocal systemd[1]: Starting Generate /run/coreos/motd...
Apr 22 09:09:44 dhub-generated-panamax.novalocal systemd[1]: Started Generate /run/coreos/motd.
Apr 22 09:10:31 dhub-generated-panamax.novalocal systemd[1]: systemd-journald.service watchdog timeout (limit 1min)!
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd[1]: Starting Flush Journal to Persistent Storage...
Apr 22 09:11:51 dhub-generated-panamax.novalocal systemd[1]: Started Flush Journal to Persistent Storage.
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="info" msg="+job serveapi(fd://)"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Using graph driver overlay"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Creating images graph"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Restored 122 elements"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Creating repository list"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Reloaded graph with 3 grants expiring at 2017-03-22 19:04:46.713978458 +0000 UTC"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="info" msg="+job init_networkdriver()"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="/usr/sbin/iptables, [--wait -C POSTROUTING -t nat -s 172.17.42.1/16 ! -o docker0 -j MASQUERADE]"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="info" msg="Listening for HTTP on fd ()"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Registering GET, /images/json"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Registering GET, /images/{name:.*}/json"
Apr 22 09:11:51 dhub-generated-panamax.novalocal dockerd[1223]: time="2015-04-22T09:11:09Z" level="debug" msg="Registering GET, /containers/ps"
```

Here is my drop-ins configuration for the systemd-journald.service unit.
```
core@dhub-generated-panamax ~ $ cat /etc/systemd/journald.conf.d/50-limit-journal-size.conf 
[Journal]
Compress=yes
SystemMaxUse=500M
SystemMaxFileSize=10M
```

**Comment by Pindar** [comment]
I'm experience the same issue with 633.1.0

**Comment by errordeveloper** [comment]
Yes, I am seeing this on _647.0.0_ too.

**Comment by errordeveloper** [comment]
```
May 15 15:06:49 kube-01 dockerd[783]: time="2015-05-15T15:06:49Z" level="info" msg="-job version() = OK (0)"
May 15 15:08:57 kube-01 systemd-journal[4924]: Permanent journal is using 24.0M (max allowed 2.7G, trying to leave 4.0G free of 23.7G available → current limit 2.7G).
May 15 15:08:57 kube-01 systemd-coredump[4921]: Detected coredump of the journal daemon itself, diverted to /var/lib/systemd/coredump/core.systemd-journal.0.8672ff8e5d4e4d1fb05eb2a648c39d2d.390.1431702501000000.xz.
May 15 15:08:57 kube-01 systemd-journal[4924]: Journal started
May 15 15:07:28 kube-01 systemd[1]: Starting Generate /run/coreos/motd...
May 15 15:07:28 kube-01 systemd[1]: Started Generate /run/coreos/motd.
May 15 15:07:49 kube-01 systemd[1]: systemd-journald.service watchdog timeout (limit 1min)!
May 15 15:08:57 kube-01 systemd[1]: Starting Flush Journal to Persistent Storage...
May 15 15:08:57 kube-01 systemd[1]: Started Flush Journal to Persistent Storage.
May 15 15:08:57 kube-01 dockerd[4925]: time="2015-05-15T15:08:44Z" level="info" msg="+job serveapi(fd://)"
```

**Comment by errordeveloper** [comment]
I am not seeing so much of problem with the `journald` crashing as such, what puzzles me is why `dockerd` gets restarted as a result...

**Comment by errordeveloper** [comment]
In fact, stoping `journald` manually causes Docker restart too:
```
May 15 15:27:05 kube-01 dockerd[4925]: time="2015-05-15T15:27:05Z" level="info" msg="-job container_inspect(motdgen.service) = ERR (1)"
May 15 15:27:05 kube-01 dockerd[4925]: time="2015-05-15T15:27:05Z" level="error" msg="Handler for GET /containers/{name:.*}/json returned error: No such container: motdgen.servic
May 15 15:27:05 kube-01 dockerd[4925]: time="2015-05-15T15:27:05Z" level="error" msg="HTTP Error: statusCode=404 No such container: motdgen.service"
May 15 15:27:05 kube-01 systemd[1]: Started Generate /run/coreos/motd.
May 15 15:27:06 kube-01 sudo[11529]: core : TTY=pts/0 ; PWD=/home/core ; USER=root ; COMMAND=/bin/systemctl stop systemd-journald
May 15 15:27:06 kube-01 polkitd[11461]: Registered Authentication Agent for unix-process:11529:193942 (system bus name :1.51 [/usr/bin/pkttyagent --notify-fd 5 --fallback], objec
May 15 15:27:06 kube-01 systemd[1]: Stopped Flush Journal to Persistent Storage.
May 15 15:27:06 kube-01 systemd[1]: Stopping Flush Journal to Persistent Storage...
May 15 15:27:06 kube-01 systemd-journal[4924]: Journal stopped
May 15 15:27:06 kube-01 systemd-journal[11535]: Permanent journal is using 40.0M (max allowed 2.7G, trying to leave 4.0G free of 23.2G available → current limit 2.7G).
May 15 15:27:06 kube-01 systemd-journald[4924]: Received SIGTERM from PID 1 (systemd).
May 15 15:27:06 kube-01 systemd-journal[11535]: Journal started
May 15 15:27:06 kube-01 polkitd[11461]: Unregistered Authentication Agent for unix-process:11529:193942 (system bus name :1.51, object path /org/freedesktop/PolicyKit1/Authentica
May 15 15:27:07 kube-01 systemd[1]: Started Docker Application Container Engine.
May 15 15:27:07 kube-01 systemd[1]: Starting Docker Application Container Engine...
May 15 15:27:07 kube-01 dockerd[11539]: time="2015-05-15T15:27:07Z" level="info" msg="+job serveapi(fd://)"
May 15 15:27:07 kube-01 dockerd[11539]: time="2015-05-15T15:27:07Z" level="info" msg="Listening for HTTP on fd ()"
May 15 15:27:07 kube-01 dockerd[11539]: time="2015-05-15T15:27:07Z" level="info" msg="+job init_networkdriver()"
May 15 15:27:07 kube-01 dockerd[11539]: time="2015-05-15T15:27:07Z" level="info" msg="-job init_networkdriver() = OK (0)"
May 15 15:27:08 kube-01 dockerd[11539]: time="2015-05-15T15:27:08Z" level="info" msg="Loading containers: start."
```

**Comment by ferrislucas** [comment]
I am seeing this on 557.2.0 and 607.0.0 running on AWS.

**Comment by errordeveloper** [comment]
I haven't managed to reproduce it on _647.2.0_ yet, but haven't tested too extensively...

**Comment by ferrislucas** [comment]
I tried this on 647.2.0, and things got weird.  From what I can reproduce, stopping journald still causes the docker engine to restart... but it won't restart until I run "docker ps" a few times after I stop journald.

Here's how I am able to reproduce the behavior:
1) Start a docker container (I just picked some wordpress thing as an example): 
docker run -d eugeneware/docker-wordpress-nginx

2) Watch one of the container processes - this process won't stop until journald is stopped and the docker engine is queried by running "docker ps". Run this to watch one of the wordpress processes:
top -p $(ps awux | grep "php-fpm: master process" | awk '{print $2}' | head -n 1)

3) Open a new console and stop journald.  Notice the wordpress process you're watching continues to run: 
sudo systemctl stop systemd-journald

4) Wait a minute or two and then run "docker ps" two or three times.  This is when the docker engine restarts for me.  Here's the log output:

Jun 02 03:02:09 ip-172-31-34-137.ec2.internal systemd[1]: Starting Generate /run/coreos/motd...Jun 02 03:02:09 ip-172-31-34-137.ec2.internal systemd[1]: Started Generate /run/coreos/motd.
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal sudo[8874]: core : TTY=pts/2 ; PWD=/home/core ; USER=root ; COMMAND=/bin/systemctl stop systemd-journald
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal polkitd[506]: Registered Authentication Agent for unix-process:8874:623657 (system bus name :1.105 [/usr/bin/pkttyagent --notify-fd 5 -
-fallback], object path /org/freedesktop/PolicyKit1/AuthenticationAgent, locale en_US.UTF-8)
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal systemd-journal[8267]: Journal stopped
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal systemd-journal[8880]: Permanent journal is using 16.0M (max allowed 4.0G, trying to leave 4.0G free of 29.7G available → current limit
 4.0G).
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal systemd-journald[8267]: Received SIGTERM from PID 1 (systemd).
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal systemd-journal[8880]: Journal started
Jun 02 03:02:49 ip-172-31-34-137.ec2.internal polkitd[506]: Unregistered Authentication Agent for unix-process:8874:623657 (system bus name :1.105, object path /org/freedesktop/Poli
cyKit1/AuthenticationAgent, locale en_US.UTF-8) (disconnected from bus)
Jun 02 03:03:19 ip-172-31-34-137.ec2.internal systemd[1]: Starting Generate /run/coreos/motd...
Jun 02 03:03:19 ip-172-31-34-137.ec2.internal systemd[1]: Started Generate /run/coreos/motd.

[this is where I ran "docker ps" a three times]

Jun 02 03:04:24 ip-172-31-34-137.ec2.internal systemd[1]: Starting Generate /run/coreos/motd...
Jun 02 03:04:24 ip-172-31-34-137.ec2.internal systemd[1]: Started Generate /run/coreos/motd.
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal systemd[1]: Started Docker Application Container Engine.
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal systemd[1]: Starting Docker Application Container Engine...
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:26Z" level="info" msg="+job serveapi(fd://)"
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:26Z" level="info" msg="Listening for HTTP on fd ()"
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:26Z" level="info" msg="+job init_networkdriver()"
Jun 02 03:04:26 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:26Z" level="info" msg="-job init_networkdriver() = OK (0)"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="Loading containers: start."
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: ......time="2015-06-02T03:04:27Z" level="error" msg="Failed to load container 5c3a16cf92e613cc0a56bc9b6e83362ee820b330bb
bc4d38c780ce16849962fe: open /var/lib/docker/containers/5c3a16cf92e613cc0a56bc9b6e83362ee820b330bbbc4d38c780ce16849962fe/config.json: no such file or directory"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: .........................
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="Loading containers: done."
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="docker daemon: 1.5.0 a8a31ef-dirty; execdriver: native-0.2; graphdriver: o
verlay"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="+job acceptconnections()"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="-job acceptconnections() = OK (0)"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="GET /v1.17/containers/json"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="+job containers()"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal dockerd[8925]: time="2015-06-02T03:04:27Z" level="info" msg="-job containers() = OK (0)"
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal kernel: docker0: port 1(veth810102d) entered disabled state
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal systemd-networkd[451]: veth810102d     : lost carrier
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal kernel: device veth810102d left promiscuous mode
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal kernel: docker0: port 1(veth810102d) entered disabled state
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal systemd-networkd[451]: veth810102d     : could not find udev device: No such device
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal systemd-networkd[451]: could not add new link: No such device
Jun 02 03:04:27 ip-172-31-34-137.ec2.internal systemd-networkd[451]: docker0         : lost carrier
Jun 02 03:04:28 ip-172-31-34-137.ec2.internal ntpd[449]: Deleting interface #40 veth810102d, fe80::941e:34ff:fecc:c7e3%69#123, interface stats: received=0, sent=0, dropped=0, active
_time=405 secs

**Comment by errordeveloper** [comment]
> ...but it won't restart until I run `docker ps` a few times after I stop journald.

Yes, I noticed that too, but it's just due to socket activation.

**Comment by ferrislucas** [comment]
Hmmmm.  I can't say that I'm super familiar with socket activation, but my understanding is that without explicitly defining .socket unit files, socket activation isn't a thing.  Maybe I'm wrong?  I don't have any .socket unit files.  I've defined various services using .service unit files which specify port mappings for their containers using the docker command line -p argument.  

Example: 
```
ExecStart=/usr/bin/docker run -p 80:80 example/image
```

Regardless, I've seen this docker engine restart issue happen in production a number of times now, and I can always trace it back to journald.  Does anyone know if this is a CoreOS issue specifically, or is this a problem I would see with any OS running docker through systemd?  Is this the proper place to report the issue?  How can I help find a solution?

**Comment by errordeveloper** [comment]
If you take a look at `systemctl cat docker.service` you will see that it refers to `docker.socket`, which is provided by CoreOS.

- [`docker.service`](https://github.com/coreos/coreos-overlay/blob/master/app-emulation/docker/files/docker.service)
- [`docker.socket`](https://github.com/coreos/coreos-overlay/blob/master/app-emulation/docker/files/docker.socket)

**Comment by errordeveloper** [comment]
It does sound like a serious issue indeed. It doesn't seem like Docker's fault to me, instead it could be more of like a really weird systemd-related issue.

**Comment by coverprice** [comment]
We've seen the same thing with Core 681.0.0. It's not just Docker restarting - it's etcd and fleet too (and probably others). systemd-journald falls over and then everything restarts.

**Comment by m1no** [comment]
We experienced the exact same thing just now :( Journal crashed and core dumped and then docker restarted.

```
Jun 25 16:49:12 sv01 systemd-journal[21528]: Permanent journal is using 1.6G (max allowed 1.7G, trying to leave 2.6G free of 2.7G available → current limit 1.6G).
Jun 25 16:49:12 sv01 systemd[1]: systemd-journald.service watchdog timeout (limit 1min)!
Jun 25 16:49:12 sv01 systemd[1]: systemd-journald.service: main process exited, code=dumped, status=6/ABRT
Jun 25 16:49:12 sv01 systemd[1]: Unit systemd-journald.service entered failed state.
Jun 25 16:49:12 sv01 systemd[1]: systemd-journald.service failed.
Jun 25 16:49:12 sv01 systemd[1]: systemd-journald.service has no holdoff time, scheduling restart.
Jun 25 16:49:12 sv01 systemd[1]: Stopping Flush Journal to Persistent Storage...
Jun 25 16:49:12 sv01 systemd[1]: Stopping Journal Service...
Jun 25 16:49:12 sv01 systemd[1]: Listening on Journal Audit Socket.
Jun 25 16:49:12 sv01 systemd[1]: Starting Journal Service...
Jun 25 16:49:40 sv01 systemd[1]: Starting Docker Application Container Engine...
Jun 25 16:49:40 sv01 systemd[1]: Started Docker Application Container Engine.
```

**Comment by pwaller** [comment]
I'm experiencing a similar set of problems on alpha 723.0.0. I have observed journald making coredumps, too, as well as the journal getting corrupted - causing `journalctl --follow` to exit and `journalctl` to show truncated logs. Since there are problems with the logging, it's hard to diagnose what the underlying causes are.

In my case, I think I'm hitting an OOM condition and the OOM killer is coming into play (due to a bug in our application). What I speculate might be happening is that the OOM killer is somehow managing to kill journald, or something along these lines. The problem is, there is nothing in the logs explaining why journald restarted (other than the presence of other OOMs indicating that we are somewhat regularly hitting OOM).

There is some more strange behaviour with the system in this state. systemd starts generally misbehaving:

```
Jun 26 04:29:03 ip-10-1-63-6.eu-west-1.compute.internal motdgen[5604]: Failed to list units: Connection timed out
Jun 26 04:29:03 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: motdgen.service: Main process exited, code=exited, status=1/FAILURE
Jun 26 04:29:03 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Failed to start Generate /run/coreos/motd.
```

Some time later I tried asking systemd for its status and restarting some services.

Doing a `systemctl status`:

```
Jun 26 08:14:30 ip-10-1-63-6.eu-west-1.compute.internal sudo[6990]: pwaller : TTY=pts/0 ; PWD=/home/pwaller ; USER=root ; COMMAND=/bin/systemctl status
Jun 26 08:14:30 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:31 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:32 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:33 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:35 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:36 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
Jun 26 08:14:37 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
```

Restarting a stopped service:

```
Jun 26 08:20:29 ip-10-1-63-6.eu-west-1.compute.internal sudo[7064]: pwaller : TTY=pts/0 ; PWD=/home/pwaller ; USER=root ; COMMAND=/bin/systemctl start tiny-ssl
Jun 26 08:20:31 ip-10-1-63-6.eu-west-1.compute.internal systemd[1]: Looping too fast. Throttling execution a little.
[ the same line above repeated once per second ]
```

I observe that while tiny-ssl was starting, it was not possible to SSH into the machine. I aborted the `sudo systemctl start tiny-ssl` and then SSH resumed working.

The full log contains sensitive information so I can't make it public. If there is something specific I should look for inside it, please let me know.

**Comment by pwaller** [comment]
I've just spotted this on the EC2 console:

```
[24682.745684] systemd-journald[403]: Failed to write entry (18 items, 612 bytes), ignoring: Bad address
```

and `journalctl` shows truncated logs with the message:

```
Failed to get journal fields: Bad message
```

**Comment by cdwertmann** [comment]
Same problem here on CoreOS 717.0.0. 
```Jul 13 00:32:01 mysql-master.novalocal systemd-coredump[16711]: Detected coredump of the journal daemon itself, diverted to /var/lib/systemd/coredump/core.systemd-journal.0.45758534aa684ba6883d8a24a946e03e.13008.1436711519000000.xz.```
Docker daemon is restarted, which in turn restarts all containers on the host.

I'm currently testing a setup with reduced log retention time for journald, hoping that it may reduce the occurrence of journald crashes.

**Comment by Pindar** [comment]
@cdwertmann My oberservation is that less logging leads to almost no crashes

**Comment by m1no** [comment]
@Pindar could you please say how you did that?

**Comment by pwaller** [comment]
Is it possible that this journald update fixes the problem? https://github.com/systemd/systemd/pull/695

If coreos provide an image with that in it (or tell me how to make one), I can test, since I'm able to reproduce it fairly easily.

**Comment by ferrislucas** [comment]
for anyone interested in a workaround, i threw in a periodic vacuum maybe 6 weeks ago, and i haven't seen a crash since:


here's the service:
```
[Unit]
Description=journald log trimmer

[Service]
TimeoutStartSec=0
ExecStart=/usr/bin/journalctl --vacuum-size=3G

[Install]
WantedBy=multi-user.target
```

here's the timer:
```
[Unit]
Description=logtrim service timer

[Timer]
OnCalendar=*-*-* 07:00:00
Persistent=true

[Install]
WantedBy=timers.target
```


send beer.

**Comment by danielschonfeld** [comment]
+1 on CoreOS Stable - 766.4.0 and 766.3.0

**Comment by schonfeld** [comment]
@ferrislucas what made you choose 3gb as your max size? We're seeing this problem happen right now when our logs are 2.6-2.7gb. How big were your log files?

**Comment by ferrislucas** [comment]
@schonfeld honestly, i picked 3gb because it's less than 4gb which i believe is the default limit.  we were hitting the 4gb limit, and that seemed to be what was causing the core dump.  my thought was that by not allowing the logs to reach the limit (in our case 4gb) then the issue would be mitigated.  

that seems to be working for us, but this certainly is not a real fix... only a mitigation.  one could argue that the 4gb limit (or whatever limit is defined on your systems) could easily be hit in between runs of the log trimming service if the system had enough logging activity.

let me know if i can shed any more light on this.  i still don't know if this is even a coreos issue or a systemd/journald issue.  i'll post here if i find that the mitigation is not working for us... so far so good.

**Comment by schonfeld** [comment]
@ferrislucas ah great, that makes sense! For us, it defaults to 2.8gb (file system is 28gb), which explains that discrepancy. This bug has caused us a lot of heartache in production... Glad to see we're not alone on this. We'll apply your workaround tonight, and keep this thread posted on any findings.

Thanks for your help!

**Comment by danielschonfeld** [comment]
Unfortunately vacuuming didn't solve the problem for us.  We've just had another crash today and just vacuumed last night with the file being half of the 10% max allowed.

We have noticed tho what seems like a memory leak in systemd-journald as each machine of ours now have a different memory size showing for this service some as high as 1GB.

I'm at a place with some bad internet at the moment but managed to create this garbage output flooding service run on both a CoreOS 766.4.0 and an Ubuntu 15.10 machines.  While a lot of the output is suppressed by journald the memory size on the CoreOS machine keeps growing slowly... on Ubuntu however it does not and stays constant.

```
[Unit]
Description=Flooder

[Service]
ExecStart=/bin/bash -c "while true; do find /; done"

[Install]
WantedBy=multi-user.target
```

**Comment by mischief** [comment]
hi @danielschonfeld, i've tried reproducing this on both CoreOS stable (766.4.0) and CoreOS alpha (835.1.0) without any similar results.

i'm running `sudo systemd-run /bin/bash -c "while true; do find /usr 2>/dev/null; done"` with `RateLimitInterval=0` in /etc/systemd/journald.conf for maximum spammyness. unfortunately, i see systemd-journald going between 10M to 100M memory usage and correctly garbage collecting on regular intervals, and no coredumps produced during the test.

any other tips to reproduce this?

**Comment by mischief** [comment]
@danielschonfeld how long did it take for your test to produce the growing memory issue in journald?

**Comment by danielschonfeld** [comment]
@mischief unfortunately i do not have any better tips.  I know i'm using a relatively "bare" installation of 766.4.0 which has some kubernetes master components running on it (I just created this VM yesterday).  Then I run the service file above and pretty much right away you start seeing `Memory` grow when you do `systemctl status systemd-journald`

About 30-40 mins after I started it the system got stuck and had to reboot it.  However, unlike what happens in our production machines when i tried finding the coredump after the reboot by doing `journalctl -t systemd-coredump` I got back the following:

```
$ journalctl -b -u systemd-coredump
Failed to get journal fields: Cannot assign requested address
-- Logs begin at Wed 2015-10-21 23:26:04 UTC, end at Fri 2015-10-23 01:09:41 UTC. --
```

**Comment by mischief** [comment]
@danielschonfeld i think you will want to use `coredumpctl` to manipulate coredumps.

**Comment by danielschonfeld** [comment]
@mischief im running it again as we speak... are there any conf files you want me to change so we can glean more info when it crashes?

**Comment by danielschonfeld** [comment]
i can't seem to induce a coredump again but i do have the coredump file in one of my production machines that experienced this problem.  i do not know why `coredumpctl list` shows nothing but the file def exists.  it's relatively small too and I think not sure the problem was during fsync.

**Comment by schonfeld** [comment]
@mischief i think we're seeing something very similar to systemd/systemd#153. I'm not sure what kind of machine you're testing on, but we (@danielschonfeld and I) are on an Azure VM. I think what's happening here is that during busy machine load times, journald is being blocked on fsync(), which in turn means that it's not sending heart beats to systemd. After 1m (journald.service's WatchdogSec default), watchdog sends a SIGBART to journald and kills it (thus explaining the coredump). Because of this, systemd just recently upped their default watchdog timeout to 3m (systemd/systemd/pull/1408).

What're your thoughts on this?

**Comment by poettering** [comment]
(not that my opinion would really matter on this, but my recommendation for coreos would be to enable the minimal debug symbol stuff we have in fedora nowadays. i.e. not do the full debug symbol stuff, but just the compressed mini version, i.e. this: https://fedoraproject.org/wiki/Features/MiniDebugInfo -- it makes life so much easier, and systemd's coredump support can read that just fine and generate stacktraces from it)

**Comment by crawford** [comment]
(@poettering I think that is a good idea. We'll look into adapting fedora's scripts for our build system.)

**Comment by mischief** [comment]
@cgeoffroy are you still able to reproduce the problem?

@schonfeld any updates wrt watchdog timeouts?

**Comment by trenton42** [comment]
Any progress on this? We are getting hit by this nearly a year later.

**Comment by vcaputo** [comment]
TL;DR
- SyncIntervalSec may be used to reduce the duration of the fsyncs which cause these watchdog failures.  The default is 5 minutes, try reducing to 30 seconds or less if you're experiencing problems.
- (System|Runtime)MaxFileSize sets upper bound of journal size **and** *apparent* memory use of the systemd-journald process.

@trenton42 We are working with upstream systemd to make these fsync calls in journald asynchronous so they do not block the event loop and cause these watchdog-induced coredumps.  It's non-trivial stuff, as it's not some simple bug to go fix, it's an architectural deficiency with the current implementation.

One thing you can do to reduce the writeback burden of these fsync calls (and thus their duration) is set `SyncIntervalSec` in the journald.conf file to something lower than the default of 5 minutes, try setting it to 30 seconds for example.  That may be a sufficient workaround until systemd-journald no longer has this issue.

With regard to the growing memory use of the systemd-journald process; all journal IO in systemd-journald is done today via mmap() windows.  This applies to Ubuntu, CoreOS, and all distributions utilizing systemd-journald alike.  These windows are cached by systemd-journald to improve performance, and since they are file-backed, they are considered "shared", and you should see in programs like `top` that the "resident" memory size of systemd-journald closely reflects the "shared" memory size.  Though it appears to the casual observer systemd-journald's memory use is increasing like a memory leak, this is mostly "shared" with the journal file, which makes the memory reclaimable by the kernel whenever memory is needed, because any dirty pages may be synced to the journal file to make the pages available.  When systemd-journald rotates a journal file the associated cached mmap windows are then freed and the "resident" memory size should drop by the size of the rotated journal file, since the new one starts empty, for the cycle to repeat itself.

What you can take away from this is that the upper bound of systemd-journald's apparent memory use is dictated by the maximum journal size.  If you try comparing behaviours across distributions, ensure you're using the same journal sizes, and you must also be producing sufficient log traffic to cause journal growth up to point of rotation to meaningfully compare the high-water marks.

**Comment by trenton42** [comment]
Thanks for the details @vcaputo! The real issue for us is that every other unit restarts when this happens. We will try the mitigation strategies here in the mean time to see if that works.

Thanks again for the explanation!

**Comment by vcaputo** [comment]
I bring good news; with https://github.com/systemd/systemd/pull/2650 merged upstream these watchdog timeouts should be substantially eliminated.

This issue will be updated shortly with our plan for how and when this lands in CoreOS.

**Comment by trenton42** [comment]
:clap: :+1: Thanks @vcaputo!

**Comment by Pindar** [comment]
:+1: :clap: :tada:

**Comment by lazarin** [comment]
thanks  :+1:

**Comment by mischief** [comment]
@vcaputo should https://github.com/systemd/systemd/pull/2650 be backported as part of updating to v229?

**Comment by mischief** [comment]
@vcaputo did https://github.com/coreos/coreos-overlay/pull/1835 fix this?

**Comment by vcaputo** [comment]
@mischief probably, at least some of the occurrences landing in this issue.

**Comment by mischief** [comment]
this original issue here should be fixed in the latest alpha (991.0.0). if not, leave a comment here.

for other issues that are not the same as the original, please file an new issue if it is still a problem.

**Comment by llarsson** [comment]
Is this fixed in the stable version at time of writing, which is 899.13.0?

I mean, it's nice that the alpha seems to have it fixed, but what about the stable version? I and my client are experiencing this bug on the old stable version, 835.13.0 and need to know if upgrading will fix it.

**Comment by agend07** [comment]
@llarsson happend to me today on 899.13.0 - so seems not to be fixed on stable

**Comment by balboah** [comment]
@cgeoffroy so how did you actually manage to get the drop-in configuration loaded? When I try to do that (to configure volatile storage to avoid watchdogs and core dumps related to disk IO) and issue a restart of journald to make it load the new config; it prevents the rest of the cloud init from finishing. There seems to be some dependency between it and journald

**Comment by crawford** [comment]
@balboah it depends on where you are running CoreOS, but you might be able to use Ignition instead. That utility runs long before cloudinit and was specifically built to solve this class of problems.

**Comment by balboah** [comment]
@crawford ah great, that's probably what I need. Thanks
